VimeoUpload/Upload/Descriptor System/New Upload (Private)/UploadDescriptor.swift

```diff
@@ -209,4 +209,21 @@ open class UploadDescriptor: ProgressDescriptor, VideoDescriptor
         
         super.encode(with: aCoder)
     }
+
+    // MARK: - Helper Methods
+    
+    internal func uploadLink(from video: VIMVideo) -> String?
+    {
+        guard let uploadLink = video.upload?.uploadLink else
```

Comment:
For this logic, I know it's unlikely that we'll get back the wrong link here but I think I'd prefer asserting that given the value of `video.upload.approach`, we look for the link in the right place. So I'm suggesting the code would say - given gcs approach, let's get the link of that object, if it's not there, assert. Otherwise if approach is streaming, get the link off upload object, 
 otherwise assert and ultimately return nil.